src/backend/distributed/planner/distributed_planner.c

```diff
@@ -115,7 +115,6 @@ static void AdjustReadIntermediateResultsCostInternal(RelOptInfo *relOptInfo,
 static List * OuterPlanParamsList(PlannerInfo *root);
```

Comment:
we should add all the tests mentioned in https://github.com/citusdata/citus/issues/4094 (if not more)

Reply:
added in 6e99005, now thinking of a bit more complicated regression tests

Reply:
we should consider adding all the examples listed in https://github.com/citusdata/citus/issues/4094, especially the sqlancer one discussed on the description

Reply:
thanks, I will add those tests too

Reply:
added in https://github.com/citusdata/citus/pull/4255/commits/abe5062d32fc3a539dba3084b55ca0c713aaa644